### PR TITLE
Implement blit on commit

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #159

### CODE REVIEW

File: src/compositor/surface.c [hunk 1]
```diff
@@ -329,7 +343,22 @@ surface_commit_cb(
...
+
+static int
+sf_commit_blit(
+    void* mon,
+    void const* buf
+) {
+    struct ws_monitor* monitor = mon;
+    struct ws_buffer const* buffer = buf;
+
+    ws_buffer_blit((struct ws_buffer *) monitor->buffer, buffer);
```
Comment (neithernut) [inline src/compositor/surface.c:359]: Please use `ws_wayland_buffer_get_buffer`() instead of dereferencing.

Comment (matthiasbeyer) [inline src/compositor/surface.c:359]: You mean instead of casting?

Comment (TheNeikos) [inline src/compositor/surface.c:359]: I think you meant `&s->img_buf` on line 348?

Comment (matthiasbeyer) [inline src/compositor/surface.c:359]: @TheNeikos There are no getter for this type (`struct ws_surface`) for getting the buffer out of it. (For `img_buf` on line 348).

**Comment by neithernut** [comment]
Is this ready for merge?

**Comment by matthiasbeyer** [comment]
IMO yes, I'm waiting for reviews and/or suggestions. It is not tested yet, though.

**Comment by neithernut** [comment]
So is a lot of stuff.

**Pull request merged**